### ISSUE TITLE
Middleware: Formalise Map / Run conventions from Katana

### ISSUE THREAD

**Issue by serialseb** [owin/owin#44]
See discussion in Pull Request #41

**Comment by panesofglass** [comment]
Must we even address these? These are nice conventions, but I think we should allow for far more possibilities than these, and supporting `Map` and/or `Run` I think put us in a position of pushing a specific, non-required pattern. Let's stick with `Use`. `Map` and `Run` can be built on top of that where desired.

NOTE: if my logic does not apply for `Run`, I'll reconsider `Run`.

**Comment by Tratcher** [comment]
Map and run don't need to be official, but I think the spec needs to allow for extensions like them. E.g. Not everything must start with use or be linear pass through.

**Comment by panesofglass** [comment]
@Tratcher, yes, I agree that the spec should allow for such extensions. How do we specify that as an open-ended possibility? I assumed that, as we did originally with middleware, by leaving it open we leave it completely open to experimentation.

**Comment by serialseb** [comment]
Anything not specified is not part of the specification. The use of SHOULD in #43 would resolve this, wouldn't it?
